lib/EntryDownloader.js

```diff
@@ -13,13 +15,15 @@ var events = require("events");
 
 var EntryDownloader = (function () {
 
-    function EntryDownloader(entryId, hostname, port, applicationName, manifestTimeWindow, manifestMaxChunkCount) {
+    function EntryDownloader(entryId, hostname, port, applicationName, flavorsIds, manifestTimeWindow, manifestMaxChunkCount) {
         events.EventEmitter.call(this);
         this.entryId = entryId;
         this.manifestTimeWindow = manifestTimeWindow;
         this.manifestMaxChunkCount = manifestMaxChunkCount;
         this.flavorsDownloaders = [];
         this.masterManifestGenerator = masterManifestGeneratorCreator(entryId, hostname, port, applicationName, logger);
+        this.smilManifestObj = smilBasedManifest(entryId, hostname, port, applicationName, logger);
```

Comment:
this.manfiestGenerator= aaa ? smilBasedManifest   : masterManifestGeneratorCreator